File: dastanlib/src/main/java/com/dastanapps/dastanlib/chat/receiver/DChatStates.java
package com.dastanapps.dastanlib.chat.receiver;

/**
 * Created by DastanIqbal on 20/10/16.
 */

public class DChatStates {
    public static ConnectionState sConnectionState;
    public static LoggedInState sLoggedInState;
    public static MessageState sMessageState;

    public static ConnectionState getsConnectionState() {
        if (sConnectionState == null) {
            return ConnectionState.DISCONNECTED;
        }
        return sConnectionState;
    }

    public static LoggedInState getLoggedInState() {
        if (sLoggedInState == null) {
            return LoggedInState.LOGGED_OUT;
        }
        return sLoggedInState;
    }

    public static MessageState getMessageState() {
        if (sMessageState == null) {
            return MessageState.NONE;
        }
        return sMessageState;
    }

    public enum ConnectionState {CONNECTED, CONNECTING, DISCONNECTING, DISCONNECTED, ERROR, AUTHENTICATED;}

    public enum LoggedInState {LOGGED_IN, LOGGED_OUT;}

    public enum MessageState {SEND, RECEIVE, NONE;}
}

File: chat/src/main/java/com/dastanapps/dastanlib/chat/receiver/DChatStates.kt
package com.dastanapps.dastanlib.chat.receiver

/**
 * Created by DastanIqbal on 20/10/16.
 */

object DChatStates {
    var sConnectionState: ConnectionState? = null
    var sLoggedInState: LoggedInState? = null
    var sMessageState: MessageState? = null

    val loggedInState: LoggedInState
        get() = if (sLoggedInState == null) {
            LoggedInState.LOGGED_OUT
        } else sLoggedInState!!

    val messageState: MessageState
        get() = if (sMessageState == null) {
            MessageState.NONE
        } else sMessageState!!

    fun getsConnectionState(): ConnectionState {
        return if (sConnectionState == null) {
            ConnectionState.DISCONNECTED
        } else sConnectionState!!
    }

    enum class ConnectionState {
        CONNECTED, CONNECTING, DISCONNECTING, DISCONNECTED, ERROR, AUTHENTICATED
    }

    enum class LoggedInState {
        LOGGED_IN, LOGGED_OUT
    }

    enum class MessageState {
        SEND, RECEIVE, NONE
    }
}
